src/main/java/com/extendedclip/papi/expansion/player/PlayerExpansion.java

```diff
@@ -1,270 +1,293 @@
-            Field pingField = nmsplayer.getClass().getDeclaredField("ping");
-            return String.valueOf(pingField.getInt(nmsplayer));
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        return "0";
-    }
-
+/*
+ *
+ * Player-Expansion
+ * Copyright (C) 2018 Ryan McCarthy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ */
+
+package com.extendedclip.papi.expansion.player;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import me.clip.placeholderapi.PlaceholderAPIPlugin;
+import me.clip.placeholderapi.expansion.Configurable;
+import me.clip.placeholderapi.expansion.PlaceholderExpansion;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.entity.Player;
+
+import java.util.Date;
+
+import static com.extendedclip.papi.expansion.player.PlayerUtil.*;
+
+public class PlayerExpansion extends PlaceholderExpansion implements Configurable {
+
+    private final String VERSION = getClass().getPackage().getImplementationVersion();
+
+    private String low, medium, high;
+
+    @Override
+    public String getIdentifier() {
+        return "player";
+    }
+
+    @Override
+    public String getAuthor() {
+        return "clip";
+    }
+
+    @Override
+    public String getVersion() {
+        return VERSION;
+    }
+
+    @Override
+    public Map<String, Object> getDefaults() {
+        Map<String, Object> defaults = new HashMap<>();
+        defaults.put("ping_color.high", "&a");
+        defaults.put("ping_color.medium", "&e");
+        defaults.put("ping_color.low", "&c");
+        return defaults;
+    }
+
+    public String onRequest(OfflinePlayer player, String identifier) {
+        high = this.getString("ping_color.high", "&a");
+        medium = this.getString("ping_color.medium", "&e");
+        low = this.getString("ping_color.low", "&c");
+
+        if (identifier.startsWith("ping_")) {
+            identifier = identifier.split("ping_")[1];
+            Player t = Bukkit.getPlayer(identifier);
+            if (t != null) {
+                return getPing(t);
+            }
+            return "0";
+        }
+        if (identifier.startsWith("colored_ping_")) {
+            identifier = identifier.split("colored_ping_")[1];
```

Comment:
Same reason like in other message.
Also why target a different player here?

Reply:
ok, and cuz its a different placeholder?